charts/karmada/templates/post-install-job.yaml

```diff
@@ -50,6 +50,15 @@ spec:
         - |
           bash <<'EOF'
           set -ex
+
+          # notes: the post-install hook job may be applied before all pods ready who relied on job/secret to be deleted,
+          # actually, thoses pods may be running init-container while main-container haven't be created, so here wait for main-container created.
+          # the sub-command `sed '1d'` is to remove the title line
+          # the sub-command `grep -vE 'Running|Completed'` is to exclude the Running or Completed pods
+          while [[ $(kubectl get pod -n {{ $namespace }} | sed '1d' | grep -vE 'Running|Completed' | wc -l) > 0 ]]; do
+            echo "waiting for all pods in karmada control plane ready..."; sleep 5;
+          done
```

Comment:
Is this all about guaranteeing the static-resource job to be executed before it gets cleaned up?

Reply:
Not very accurate, main purpose is to prevent the following timing events from occurring:
1. karmada components, i.e: karmada-controller-manager, be appiled, attention please, just appiled, not running.
2. post-install-job running, and delete the `static-resource job` and `hook-job secret` quickly.
3. karmada components, i.e: karmada-controller-manager, failed with lack of `hook-job secret` or its `kubectl wait --for=condition=complete job xx-static-resource` never meets.

So, we expect the deletion of  `static-resource job` and `hook-job secret` should be delayed to when we never need them, that is, all karmada components running.

Reply:
done and tested ok